snuba/pipeline/single_query_plan_pipeline.py

```diff
@@ -1,58 +1,97 @@
+        self.__clickhouse_processors = processors
+
+    def execute(
+        self, query: LogicalQuery, settings: RequestSettings
+    ) -> ClickhouseQueryPlan:
+        self.__entity_processors.execute(query, settings)
+
+        query_plan = self.__query_plan_builder.build_plan(query, settings)
+        self.__clickhouse_processors.execute(query_plan, settings)
+        return query_plan
 
-    def execute(self) -> QueryResult:
-        # Execute entity processors
-        _execute_entity_processors(self.__request)
 
-        # Build and execute query plan
-        self.__query_plan = self.__query_plan_builder.build_plan(self.__request)
-        _execute_query_plan_processors(self.__query_plan, self.__request)
-        return self.__query_plan.execution_strategy.execute(
-            self.__query_plan.query, self.__request.settings, self.__runner
+class PreStrategyProcessingPipeline(SinglePlanProcessingPipeline):
+    """
+    Executes the query processing of a single query plan query up to the
+    plan query processors. It does not execute the db query processors
+    since they are ran by the execution strategy.
+
+    This is the processing pipeline to use when executing a simple query
+    that references a single entity and a single storage.
+    """
+
+    def __init__(self, query_plan_builder: ClickhouseQueryPlanBuilder) -> None:
+        super().__init__(
+            query_plan_builder,
+            ClickhouseProcessorsExecutor(lambda plan: plan.plan_processors),
         )
 
-    @property
-    def query_plan(self) -> ClickhouseQueryPlan:
-        return self.__query_plan
+
+class SinglePlanExecutionPipeline(QueryExecutionPipeline):
+    """
+    Executes a simple (single entity) query.
+    This relies on a PreStrategyProcessingPipeline to perform the processing
+    and then relies on the execution strategy provided by the query plan to
+    actually execute the query.
+    """
+
+    def __init__(
+        self, runner: QueryRunner, query_plan_builder: ClickhouseQueryPlanBuilder,
+    ):
+        self.__runner = runner
+        self.__processing_pipeline = PreStrategyProcessingPipeline(query_plan_builder)
+
+    def execute(self, input: Request) -> QueryResult:
+        settings = input.settings
+
+        plan = self.__processing_pipeline.execute(input.query, settings)
+        return plan.execution_strategy.execute(plan.query, settings, self.__runner)
 
 
 class SingleQueryPlanPipelineBuilder(QueryPipelineBuilder):
     def __init__(self, query_plan_builder: ClickhouseQueryPlanBuilder) -> None:
         self.__query_plan_builder = query_plan_builder
 
-    def build_pipeline(self, request: Request, runner: QueryRunner) -> QueryPipeline:
-        return SingleQueryPlanPipeline(request, runner, self.__query_plan_builder)
+    def build_execution_pipeline(
+        self, request: Request, runner: QueryRunner
+    ) -> QueryExecutionPipeline:
+        return SinglePlanExecutionPipeline(runner, self.__query_plan_builder)
```

Comment:
This is very confusing. This class has `build_execution_pipeline`, which builds a `SinglePlanExecutionPipeline` which in turn builds a `PreStrategyProcessingPipeline` which is a very thin wrapper around the `SinglePlanProcessingPipeline` and executes that as well as the buried execution strategy in the query plan. However it also has a `build_processing_pipeline` function that also returns a `PreStrategyProcessingPipeline`.

Reply:
Unfortunately the entity provides the query pipeline, in the execution case we need an SinglePlanExecutionPipeline from the pipeline builder, in the composite case we will need a QueryProcessingPipeline, thus two builder methods. We can use the builder for the execution pipeline as well, not sure how much more clear will be. Will try out.

Reply:
This should not apply anymore since neither PreStrategyProcessingPipeline nor build_processing_pipeline exist anymore.

Reply:
I'm starting to think the entity shouldn't be the one providing the query pipeline anymore (in both the single entity and join cases). Could we just pass the entity/entities being requested to the pipeline builder and have it just build the appropriate pipeline. That way were are always building the pipeline in one place, we don't need the two separate methods to be called in different scenarios.